lib/facility.rb

```diff
@@ -1,14 +1,63 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_accessor :name,
+                :address,
+                :phone,
+                :services,
+                :registered_vehicles,
+                :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    @services.include?('Vehicle Registration')
+      collect_fees(vehicle)
```

Comment:
Great work extracting this data out into separate helper methods! REally cleans up this method